### ISSUE TITLE
Wildcard handling broken with a recent commit

### ISSUE THREAD

**Issue by rderooy** [joncampbell123/dosbox-x#1475]
**Describe the bug**
Wildcard handling is broken, at least in one particular situation due to a recent commit. This effects both copy or xcopy using wildcards which should transfer multiple files, and it will only copy the first file.

This works:
- dosbox-x 0.83.0
- copying from a host volume to a HDD image
- copying from a HDD image to a host volume
- copying from a ISO image to a host volume
- copying files from one directory to another on the same disk image
- copying files from one HDD image to another HDD image

This fails:
- copying from a ISO disk image to a HDD image
![command_002](https://user-images.githubusercontent.com/17701098/79114235-0a95d200-7d83-11ea-8ba8-51c2b4eeb74b.png)

**To Reproduce**
- compile the latest dosbox-x from git
- start dosbox-x
- mount some disk images, of which one is an ISO
- copy files from the ISO to the HDD image

**Expected behavior**
wildcards to work as before

**Environment (please complete the following information):**
- Fedora 31
- DOSBox-X 0.83.1-git

**Comment by rderooy** [comment]
A few other peculiarities with the wildcard handling of the integrated DOS.

- "*" is treated like "*.*". In real DOS you need to specify *.* to specify all the files, specifying just "*" in real DOS will result in only the files with NO extension. This may result in unexpected behaviour.

- Cannot delete files with wildcards of the destination drive, unless it is the current drive

Integrated DOS
![command_004](https://user-images.githubusercontent.com/17701098/79120033-1c32a600-7d92-11ea-9f54-029833530a05.png)

Real DOS booted in DOSBox-X
![guest os_001](https://user-images.githubusercontent.com/17701098/79120009-0ae99980-7d92-11ea-9bbc-dbbb147c701d.png)

**Comment by Wengier** [comment]
So the regression mentioned in your first post only occurs when you copy from a ISO disk image to a HDD image, but not others. I have fixed this as part of the patch I posted in Issue #1474.

As for the peculiarities with the wildcard mentioned in your second post, they occurred in the earlier official 0.83.0 release too, not just the latest code, so they are completely separate issues. Nevertheless, I have fixed both of them as follows:

1: DEL \* will now only delete files with NO extension in SFN mode as in real DOS.
2: Deleting files with \*.\* wildcards should now work regardless if it is the current drive. For example: DEL B:\*.\* will delete all files in the current directory of Drive B: after confirmation.
